Add vehicle chargings and trips API endpoint

GewoonJaap opened the pull request:
This PR adds endpoints to retrieve vehicles charging sessions and the trips.

GewoonJaap commented:
Also added CURL to the Docker image to support HTTP healthchecks for the container.

flobz commented:
thanks for you contribution I will look at it.

GewoonJaap commented:
Thank you @flobz I see the pipeline fails now because I added the curl package to enable docker HTTP health checks. Should I pin curl to a version?

vineethktpla commented:
Very useful and good work @GewoonJaap . Was waiting for this for a long time and I hope it will merge soon.
This will ease my work in home assistant to track the trips. I am currently using trigger based automations to calculate and my trips and charging sessions

flobz commented on psa_car_controller/web/view/api.py:
```diff
@@ -181,6 +186,27 @@ def settings_section(section: str):
         APP.config.write_config()
     return json_response(config_section.json())
 
+@app.route('/get_vehicletrips')
```
```suggestion
@app.route('/vehicles/trips')
```
wdyt ?

flobz commented on psa_car_controller/web/view/api.py:
```diff
@@ -181,6 +186,27 @@ def settings_section(section: str):
+        return FlaskResponse(json.dumps([]), mimetype='application/json')
+
+
+@app.route('/get_vehiclechargings')
```
```suggestion
@app.route('/vehicles/chargings')
```

flobz commented on psa_car_controller/web/view/api.py:
```diff
@@ -181,6 +186,27 @@ def settings_section(section: str):
+        trips_by_vin = Trips.get_trips(Cars([car]))
+        trips = trips_by_vin[car.vin]
+        trips_as_dict = trips.get_trips_as_dict()
+        return FlaskResponse(json.dumps(trips_as_dict, default=str), mimetype='application/json')
```
Could you use jsonify like this :
```suggestion
        return jsonify(trips_as_dict)
```

GewoonJaap replied:
Yeah that looks better indeed! I just updated :)

flobz reviewed (changes requested):
Thanks for you contribution.
Could you look to my suggested changes ?

GewoonJaap commented:
> Thanks for you contribution. Could you look to my suggested changes ?

Updated the API and readme with the new endpoints and changed the JSON return method as you suggested :)

flobz commented:
lgtm thanks !
